File: YH_Toast/YH_Toast/YH_Toast.m
//
//  YH_Toast.m
//  qtz
//
//  Created by 余晗 on 12/2/15.
//  Copyright (c) 2015 tmbj. All rights reserved.
//

#import "YH_Toast.h"
#import "AppDelegate.h"
@implementation YH_Toast
+(void)showToastFromStr:(NSString *)str
{
    
    UIWindow *window = [[[UIApplication sharedApplication] windows] lastObject];

  float screenWidth=[[UIScreen mainScreen] bounds].size.width;
  float screenHeight=[[UIScreen mainScreen] bounds].size.height;
  CGSize strSize=[str sizeWithFont:[UIFont systemFontOfSize:ToastFont] constrainedToSize:CGSizeMake(screenWidth-60, 200)];
  float realViewHeight=0.f;
  if([[[UIDevice currentDevice] systemVersion] floatValue]>7.0)
  {
    realViewHeight=20.f+screenHeight;
  }
  else
  {
   realViewHeight=0.f+screenHeight;
  }
  
  UIView *contentView=[[UIView alloc] initWithFrame:CGRectMake((screenWidth-(strSize.width+20))/2, realViewHeight-strSize.height-40-20,strSize.width+20, strSize.height+20)];
  contentView.layer.cornerRadius=5.f;
  contentView.layer.shadowColor = [UIColor blackColor].CGColor;
  contentView.layer.shadowOffset = CGSizeMake(0, 0);
  contentView.layer.shadowOpacity = 0.5;
  contentView.layer.shadowRadius = 1;
  [contentView setBackgroundColor:BGColor];
  [window addSubview:contentView];

  UILabel *lbl=[[UILabel alloc] initWithFrame:CGRectMake(0, 0, contentView.frame.size.width,contentView.frame.size.height)];
  [lbl setTextAlignment:NSTextAlignmentCenter];
  [lbl setTextColor:ToastStrColor];
  [lbl setFont:[UIFont systemFontOfSize:ToastFont]];
  [lbl setText:str];
  [lbl setNumberOfLines:10];
  [contentView addSubview:lbl];
  
  [UIView animateWithDuration:3.f animations:^{
    contentView.alpha=0.f;
     } completion:^(BOOL finished) {
       [lbl removeFromSuperview];
       [contentView removeFromSuperview];
     }];
}
/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

@end

File: YH_Toast_swift/YH_Toast_swift/YH_Toast.swift
//
//  YH_Toast.swift
//  testSwift
//
//  Created by 余晗 on 1/11/16.
//  Copyright © 2016 余晗. All rights reserved.
//

import UIKit
let toastFont  : CGFloat = 12;
let toastStrColor = UIColor.whiteColor();
let BGColor = UIColor(colorLiteralRed: 39/255, green: 39/255, blue: 39/255, alpha: 1);
class YH_Toast: UIView {

    var contentView :UIView = UIView();
     func getStrSize(text:String,font:UIFont,size:CGSize) -> CGRect {
         let strAtt = [NSFontAttributeName: font]
        let needOption = NSStringDrawingOptions.UsesLineFragmentOrigin
        let rect:CGRect = text.boundingRectWithSize(size, options: needOption, attributes: strAtt, context: nil)
        return rect;
    }
    
    func showToastFromStr(str :String)
    {
        let lastWindows = UIApplication.sharedApplication().windows.last;
        
        let screenWidth  = UIScreen.mainScreen().bounds.size.width;
        let screenHeight = UIScreen.mainScreen().bounds.size.height;
        
        let strSize = getStrSize(str, font: UIFont.systemFontOfSize(toastFont), size:CGSizeMake(screenWidth-60, 200));
        
        let realViewHeight = 20 + screenHeight ;
      
      contentView.frame = CGRectMake((screenWidth-(strSize.width+20))/2, realViewHeight-strSize.height-40-20,strSize.width+20, strSize.height+20);
        contentView.layer.cornerRadius = 5;
        contentView.layer.shadowColor = UIColor.blackColor().CGColor;
        contentView.layer.shadowOffset = CGSizeMake(0, 0);
        contentView.layer.shadowOpacity = 0.5;
        contentView.layer.shadowRadius = 1;
        contentView.backgroundColor=BGColor;
        lastWindows!.addSubview(contentView);
        
        let lbl = UILabel(frame: CGRectMake(0, 0, contentView.frame.size.width,contentView.frame.size.height));
        lbl.textAlignment=NSTextAlignment.Center;
        lbl.textColor=toastStrColor;
        lbl.font = UIFont.systemFontOfSize(toastFont);
        lbl.text = str;
        lbl.numberOfLines = 10;
        contentView.addSubview(lbl);
        
        UIView.animateWithDuration(3,animations:
            { ()-> Void in
                self.contentView.alpha=0;
            },
            completion:{
                (finished:Bool) -> Void in
                self.contentView.removeFromSuperview();
                lbl.removeFromSuperview();
        })
    }

    // Only override drawRect: if you perform custom drawing.
    // An empty implementation adversely affects performance during animation.
//    override func drawRect(rect: CGRect) {
              // Drawing code
//    }


}
